addons/udes_stock_picking_batch/models/stock_picking.py

```diff
@@ -56,7 +56,8 @@ def action_assign(self):
         but the constraint on state has not proven to work correctly.
         """
         res = super().action_assign()
-        self.batch_id._compute_state()
+        if self.exists():   # This picking can get deleted after merge
```

Comment:
Found in unit tests that this picking may get unlinked after merging take place in `action_assign()`.